Source/dafny.msbuild/dafny.msbuild.csproj

```diff
@@ -21,6 +21,8 @@
         <PackageReference Include="Microsoft.Build.Framework" Version="15.1.1012" PrivateAssets="All" />
         <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.1.1012" PrivateAssets="All" />
         <PackageReference Include="System.Linq.Parallel" Version="4.3.0" PrivateAssets="All" />
+        <PackageReference Include="System.Collections.Immutable" Version="1.7.0" PrivateAssets="All" />
```

Comment:
Watch out for the difference between a dependency of dafny.msbuild itself, which shouldn’t be transitively a dependency of a project using this build tool, and one that IS intended to be like these ones. It’s possible the existing dependencies are getting picked up when they shouldn’t be. :)

Reply:
Ah, my new dependencies should probably be `IncludeAssets`

Reply:
I think you might be getting a spurious build pass here, because this project's build doesn't lock down the Dafny commit like our others. It might have built with a snapshot of Dafny before you actually made the dependency required?

Reply:
Few notes here...
1. The way this was written wasn't actually doing anything. Per .NET documentation, we should use IncludeAssets to pass these. This also wasn't actually working as expected, though, as it appeared to make assets consumable, but still required the child to reference that asset (via PackageReference).
2. The reason this was working was because we were targeting `netcoreapp3.0` in TestProject.csproj. If we target `netstandard2.0` instead, tests can't run (since it's not a runnable app), but we can resolve the dependency as long as we add a package reference to `System.Collections.Immutable`. If we don't add the package reference, this fails for the expected reason `The type or namespace name 'Immutable' does not exist in the namespace 'System.Collections'`
3. It appears `PackageReference` is required in TestProject for `System.Collections.Immutable` regardless of whether or not it's added in dafny.msbuild.csproj (or whether it's PrivateAssets/IncludeAssets in dafny.msbuild.csproj). Adding guidance.
4. I haven't experimented much with this, but

Reply:
Regarding 2, it occurs to me that it would be good to have another test project that doesn't include tests and the `xunit` dependencies. That would allow it to target `netstandard2.0` and avoid potentially bringing in the `System.Collections.Immutable` dependency via `xunit`.